Exception stack for JDBC connection failure is too much

**Describe the bug**
The exception stack trace for JDBC connection failure is way too much useless information.

**Expected behavior**
An exception clearly describing the failure with a stack trace leading to the point of failure.

**Actual behavior**
Something like this (note that the first stack trace reflects some prototype code, so don't worry about it):
```
javax.persistence.PersistenceException: [PersistenceUnit: templatePU] Unable to build Hibernate SessionFactory
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:122)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:82)
	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:70)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:79)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:54)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:125)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:73)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmTemplate.startAllPersistenceUnits(HibernateOrmTemplate.java:93)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits67.deploy_0(HibernateOrmProcessor$startPersistenceUnits67.zig:49)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits67.deploy(HibernateOrmProcessor$startPersistenceUnits67.zig:68)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:16)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:17)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:17)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:17)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:17)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:17)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ClassicExecutionHandler.run(ClassicExecutionHandler.java:17)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.RestartingExecutionHandler.run(RestartingExecutionHandler.java:28)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.SignalHandlerExecutionHandler.run(SignalHandlerExecutionHandler.java:76)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ShutdownHookExecutionHandler.run(ShutdownHookExecutionHandler.java:32)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.execution.ConfigurationHandler.run(ConfigurationHandler.java:24)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.proceed(ExecutionChain.java:44)
	at io.quarkus.runtime.generated.SystemProperties.run(SystemProperties.zig:35)
	at io.quarkus.runtime.execution.SupplierExecutionHandler.run(SupplierExecutionHandler.java:71)
	at io.quarkus.runtime.execution.ExecutionChain.run(ExecutionChain.java:57)
	at io.quarkus.runtime.execution.ExecutionChain.startAndExit(ExecutionChain.java:77)
	at io.quarkus.runtime.Quarkus.main(Quarkus.java:64)
Caused by: org.hibernate.exception.GenericJDBCException: Unable to open JDBC Connection for DDL execution
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:47)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
	at org.hibernate.resource.transaction.backend.jta.internal.DdlTransactionIsolatorJtaImpl.<init>(DdlTransactionIsolatorJtaImpl.java:62)
	at org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorBuilderImpl.buildDdlTransactionIsolator(JtaTransactionCoordinatorBuilderImpl.java:46)
	at org.hibernate.tool.schema.internal.HibernateSchemaManagementTool.getDdlTransactionIsolator(HibernateSchemaManagementTool.java:175)
	at org.hibernate.tool.schema.internal.HibernateSchemaManagementTool.buildGenerationTargets(HibernateSchemaManagementTool.java:135)
	at org.hibernate.tool.schema.internal.SchemaDropperImpl.doDrop(SchemaDropperImpl.java:110)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:145)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:73)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:309)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:80)
	... 53 more
Caused by: java.sql.SQLException: Exception while creating new connection
	at io.agroal.pool.ConnectionPool.handlerFromSharedCache(ConnectionPool.java:225)
	at io.agroal.pool.ConnectionPool.getConnection(ConnectionPool.java:146)
	at io.agroal.pool.DataSource.getConnection(DataSource.java:61)
	at io.quarkus.agroal.runtime.DataSourceProducer_ProducerMethod_createDefaultDataSource_7c487e3ef869f878aa871e917c94f4d26d5d5c56_ClientProxy.getConnection(DataSourceProducer_ProducerMethod_createDefaultDataSource_7c487e3ef869f878aa871e917c94f4d26d5d5c56_ClientProxy.zig:102)
	at org.hibernate.engine.jdbc.connections.internal.DatasourceConnectionProviderImpl.getConnection(DatasourceConnectionProviderImpl.java:122)
	at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator$ConnectionProviderJdbcConnectionAccess.obtainConnection(JdbcEnvironmentInitiator.java:180)
	at org.hibernate.resource.transaction.backend.jta.internal.DdlTransactionIsolatorJtaImpl.<init>(DdlTransactionIsolatorJtaImpl.java:59)
	... 62 more
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Exception while creating new connection
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at io.agroal.pool.ConnectionPool.handlerFromSharedCache(ConnectionPool.java:193)
	... 68 more
Caused by: java.lang.RuntimeException: Exception while creating new connection
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:357)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:337)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:63)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1146)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.h2.jdbc.JdbcSQLException: Connection is broken: "java.net.ConnectException: Connection refused (Connection refused): localhost" [90067-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.engine.SessionRemote.connectServer(SessionRemote.java:451)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:332)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:178)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:348)
	... 6 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.h2.util.NetUtils.createSocket(NetUtils.java:103)
	at org.h2.util.NetUtils.createSocket(NetUtils.java:83)
	at org.h2.engine.SessionRemote.initTransfer(SessionRemote.java:114)
	at org.h2.engine.SessionRemote.connectServer(SessionRemote.java:447)
	... 12 more
```


**To Reproduce**
Run the main integration test without setting up a database instance.

--- a/bom/pom.xml
+++ b/bom/pom.xml
@@ -96,7 +96,7 @@
         <hibernate-search.version>6.0.0.Alpha6</hibernate-search.version>
         <narayana.version>5.9.3.Final</narayana.version>
         <jboss-transaction-api_1.2_spec.version>1.1.1.Final</jboss-transaction-api_1.2_spec.version>
-        <agroal.version>1.4</agroal.version>
+        <agroal.version>1.5</agroal.version>
         <jboss-transaction-spi.version>7.6.0.Final</jboss-transaction-spi.version>
         <javax.persistence-api.version>2.2</javax.persistence-api.version>
         <elasticsearch-rest-client.version>7.0.1</elasticsearch-rest-client.version>

